ts/Accessibility/Components/LegendComponent.ts

```diff
@@ -535,21 +535,32 @@ class LegendComponent extends AccessibilityComponent {
             response = keyboardNavigationHandler.response,
             chart = this.chart,
             a11yOptions = chart.options.accessibility,
+            wrapAround = a11yOptions.keyboardNavigation.wrapAround,
             numItems = chart.legend.allItems.length,
-            direction = (keyCode === keys.left || keyCode === keys.up) ? -1 : 1;
+            direction = (
+                keyCode === keys.left || keyCode === keys.up
+            ) ? -1 : 1,
+            highlightedLegendItemIx = this.highlightedLegendItemIx,
+            rawDirectionalInput = highlightedLegendItemIx + direction,
+            adjustedDirection = ((
+                rawDirectionalInput > -1 &&
+                rawDirectionalInput < numItems
+            ) ? direction : (
+                    wrapAround && (
+                        (numItems - 1) * (direction > 0 ? -1 : 1)
+                    ) || 0
+                )
+            ),
+            res = chart.highlightLegendItem(
+                highlightedLegendItemIx + adjustedDirection
+            ) || void 0;
 
-        const res = chart.highlightLegendItem(
-            this.highlightedLegendItemIx + direction
-        );
         if (res) {
-            this.highlightedLegendItemIx += direction;
+            this.highlightedLegendItemIx += adjustedDirection;
             return response.success;
         }
 
-        if (
-            numItems > 1 &&
-            a11yOptions.keyboardNavigation.wrapAround
-        ) {
+        if (wrapAround && numItems > 1) {
             keyboardNavigationHandler.init(direction);
             return response.success;
```

Comment:
Nitpick:
The code structure could be improved - DRY. Now there's:
```
        if (res) {
            this.highlightedLegendItemIx += adjustedDirection;
            return response.success;
        }

        if (wrapAround && numItems > 1) {
            keyboardNavigationHandler.init(direction);
            return response.success;
        }

        return response.success;
```

Always `return response.success`. How about something like `if(A) { B; } else if (C) { D; } return S;`

Reply:
Good suggestion!

Also regarding DRY-principle, a thought I had while solving this bug; 

Just as a basic example, I believe in this function, `chart` is referenced twice during var-declaration at the start of the function, but now with the de-structuring syntax, we could easily limit this to one such reference. 

This will really break the old syntactic style of var-declaration, but it will limit property lookup, and thus, in theory improve performance if one can look up a property only once. 

Since there is a lot of property lookups in a big project like HC, should we move towards destructuring?